pkg/actors/actors.go

```diff
@@ -951,11 +951,12 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
 	stop := make(chan bool, 1)
 	a.activeTimers.Store(timerKey, stop)
 
-	go func(stop chan (bool), req *CreateTimerRequest) {
-		time.Sleep(dueTime)
```

Comment:
Sleep on dueTime is important because it is the expected behavior for the dueTime parameter of a timer. time.After() will just compare now with the passed parameter. Why does this change keep the existing behavior?

Reply:
Thank you for your reply. But if the timer is terminated prematurely, it is meaningless to wait for sleep to return, right?

Reply:
This change makes sense to me however I think the `default: break` would need to be removed. Correct?

Reply:
I agree with @pkedy's opinion. The `default: break` should be removed.

Reply:
> This change makes sense to me however I think the `default: break` would need to be removed. Correct?

Sorry, my bad. I will remove `default: break`.

Reply:
@artursouza time.After() will create a timer and return channel, so I think it can replace the sleep code. 

```go
// After waits for the duration to elapse and then sends the current time
// on the returned channel.
// It is equivalent to NewTimer(d).C.
// The underlying Timer is not recovered by the garbage collector
// until the timer fires. If efficiency is a concern, use NewTimer
// instead and call Timer.Stop if the timer is no longer needed.
func After(d Duration) <-chan Time {
	return NewTimer(d).C
}
```

@CodingSinger Should we add a checking in stop case that if we stop before duetime, we should stop the time. like this:

```
		// Check if timer is still active
		t := time.NewTimer(dueTime)
		select {
		case <-t.C:
			log.Debugf("Time: %v with parameters: DueTime: %v, Period: %v, Data: %v has been overdue.", timerKey, req.DueTime, req.Period, req.Data)
			t = nil
			break
		case <-stop:
			log.Infof("Time: %v with parameters: DueTime: %v, Period: %v, Data: %v has been deleted.", timerKey, req.DueTime, req.Period, req.Data)
			if t != nil {
				t.Stop()
				t = nil
			}
			return
		}
```

Reply:
@skyao Thank you very much for your advice.

Reply:
@skyao please help me approve  the workflows.